From 25e2210db7827c50da7f0d6cdca7840b6c3bd804 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 01 Jun 2023 16:32:41 +0300
Subject: [PATCH] [SW-146646] gaudi3: skip hash verification based on sim mode

Currently, the asic target is used to run FW flow on simulator. There
are certain configurations such as d2d and HBM trainings which needs
to be skipped on simulator. Also to speed up the boot flow certain
time consuming operations such as hash verification of images needs
to be skipped.

This patch add changes to skip preboot hash verification based on
SIM_MODE fuse bit.

Change-Id: I794b8af9a8c1402ea21c929e1f1622ff6856edd5
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 20fc801..461423a 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -62,6 +62,7 @@
 #include <pif_ts.h>
 #include <hl_pif_arc_if.h>
 #include <hl_fw_if.h>
+#include <gaudi3_efuse.h>
 
 #define PIF_ARC_NUM_OF_MEM	2
 
@@ -487,6 +488,28 @@
 	return !val;
 }
 
+bool hl_is_sim_mode(void)
+{
+	u32 val;
+	int err;
+
+	err = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &val);
+	if (err)
+		return false;
+
+	if (val & EFUSE_EOM_MASK)
+		return false;
+
+	err = preboot_fuse_read(EFUSE_SYSCFG2_OFFSET, &val);
+	if (err)
+		return false;
+
+	if (val & EFUSE_SIM_MODE)
+		return true;
+
+	return false;
+}
+
 #define GAUDI3_EFUSE_READ_RANGE_ALL	0xff00
 
 void device_init_efuse(void)
@@ -711,6 +734,9 @@
 	unsigned int num_crc;
 	int ret;
 
+	if (hl_is_sim_mode())
+		return true;
+
 	num_crc = ppboot_get_num_crc(desc->img_size);
 	actual_size = desc->img_size - (num_crc * CRC32_SIZE);
 	addr = ppboot_get_arc_addr(desc->loadaddr);
